### PR TITLE
CONSOLE-2472: Add initial quick starts

### PR DESCRIPTION
**Pull request by spadgett**
This PR contributes the initial out-of-the-box quick start custom resources and adds a brief README discussing the contribution process.

[CONSOLE-2472](https://issues.redhat.com/browse/CONSOLE-2472)

/assign @jhadvig

### CODE REVIEW

**Comment by spadgett** [comment]
cc @christianvogt @rohitkrai03 @alimobrem

File: quickstarts/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Details about the API are also covered in the
+[quick start enhancement proposal](https://github.com/openshift/enhancements/blob/master/enhancements/console/quick-starts.md).
+
+## Updating Quick Starts in Previous Releases
+
+The console-operator repo has branches for each OpenShift release. The `master`
+branch tracks the next unreleased minor (`y` version) of OpenShift. Releases
+that have already shipped are tracked through branches like `release-4.6`. If
+you need to backport a quick start change to a previous release, you will need
+to a Bugzilla bug. The `/cherry-pick` bot command will automatically create a
```
Comment (christianvogt) [inline quickstarts/README.md:39]:
```suggestion
a Bugzilla bug. The `/cherry-pick` bot command will automatically create a
```

Comment (spadgett) [reply quickstarts/README.md]: Thanks, I pushed a fix for this typo

**Comment by spadgett** [comment]
/hold

The manifests are missing the cluster profile annotations.

**Comment by spadgett** [comment]
/hold cancel

Updated with the annotations.

**Comment by spadgett** [comment]
@beanh66 FYI

File: quickstarts/explore-pipeline-quickstart.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  description: Install the OpenShift® Pipelines Operator to build Pipelines using
+    Tekton.
+  prerequisites: ['']
```
Comment (christianvogt) [inline quickstarts/explore-pipeline-quickstart.yaml:16]:
```suggestion
```

File: quickstarts/explore-serverless-quickstart.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  description: Install the OpenShift Serverless Operator to deploy stateless, event-trigger-based
+    applications.
+  prerequisites:
+    - ''
```
Comment (christianvogt) [inline quickstarts/explore-serverless-quickstart.yaml:17]:
```suggestion
```

File: quickstarts/sample-application-quickstart.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+spec:
+  version: 4.7
+  displayName: Getting started with a sample
+  durationMinutes: 10
+  icon: ''
+  description: Is this the first time you’ve used OpenShift? Let's start with a simple
+    sample app to learn the basics.
+  prerequisites: ['']
```
Comment (christianvogt) [inline quickstarts/sample-application-quickstart.yaml:16]:
```suggestion
```

File: quickstarts/add-healthchecks-quickstart.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  name: add-healthchecks
+  annotations:
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+spec:
+  version: 4.7
+  displayName: Adding health checks to your sample application
+  durationMinutes: 10
+  icon: ''
```
Comment (christianvogt) [inline quickstarts/add-healthchecks-quickstart.yaml:13]:
```suggestion
```

File: quickstarts/monitor-sampleapp-quickstart.yaml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  name: monitor-sampleapp
+  annotations:
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+spec:
+  version: 4.7
+  displayName: Monitoring your sample application
+  durationMinutes: 10
+  icon: ''
```
Comment (christianvogt) [inline quickstarts/monitor-sampleapp-quickstart.yaml:13]:
```suggestion
```

File: quickstarts/monitor-sampleapp-quickstart.yaml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      review:
+        instructions: |
+          #### Verify you can see the chart associated with the query:
+          Do you see a chart displayed with filesystem usage for your project?  Note: select **Custom Query** from the dropdown to create and run a custom query utilizing PromQL.
+        failedTaskHelp: This task isn’t verified yet. Try the task again.
+      summary:
+        success: You have learned how to run a query!
+        failed: Try the steps again.
+  conclusion: You have learned how to access workload monitoring and metrics!
+  nextQuickStart: ['']
```
Comment (christianvogt) [inline quickstarts/monitor-sampleapp-quickstart.yaml:70]:
```suggestion
```

File: quickstarts/sample-application-quickstart.yaml [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  name: sample-application
+  annotations:
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+spec:
+  version: 4.7
+  displayName: Getting started with a sample
+  durationMinutes: 10
+  icon: ''
```
Comment (christianvogt) [inline quickstarts/sample-application-quickstart.yaml:13]:
```suggestion
```

**Review by christianvogt — COMMENTED** [review]
@spadgett I know this isn't your doing as the yaml files were provided. But we should remove the the empty properties.

File: quickstarts/sample-application-quickstart.yaml [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+          #### To verify your sample application is running:
+          1. Make sure you are in the new tab.
+          2. Does the page have a **Welcome to your Node.js application on OpenShift** title?
+        failedTaskHelp: This task isn’t verified yet. Try the task again.
+      summary:
+        success: You have run your sample app!
+        failed: Try the steps again.
+  conclusion: Your sample application is deployed and ready! To add health checks
+    to your sample app, take the **Adding health checks to your sample application**
+    quick start
```
Comment (jhadvig) [inline quickstarts/sample-application-quickstart.yaml:100]:
```suggestion
    quick start.
```

File: quickstarts/explore-serverless-quickstart.yaml [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+  description: Install the OpenShift Serverless Operator to deploy stateless, event-trigger-based
```
Comment (jhadvig) [inline quickstarts/explore-serverless-quickstart.yaml:14]:
```suggestion
  description: Install the OpenShift® Serverless Operator to deploy stateless, event-trigger-based
```

**Review by jhadvig — CHANGES_REQUESTED** [review]
Two small nits.
Also want to ask if we shall differentiate between `OpenShift®` and `OpenShift` cause the usage is a bit strange.

File: quickstarts/explore-serverless-quickstart.yaml (hunk 10 above)
Comment (christianvogt) [reply quickstarts/explore-serverless-quickstart.yaml:14]: There are so many instances of `OpenShift` and a handful instances of `OpenShift®`.
I don't think it's worth getting into the semantics in this PR and content writers can revise them later.
I reached out to UX and their guidance was:
> We chatted about this with CCS and they reached out to legal who recommended marking the first appearance of trademarked terms

**Comment by spadgett** [comment]
Thanks @christianvogt and @jhadvig. I just pushed some the recommended updates.

File: quickstarts/explore-serverless-quickstart.yaml (hunk 10 above)
Comment (beaumorley) [reply quickstarts/explore-serverless-quickstart.yaml:14]: We'll look at the copy when it is all migrated and merged. May have an update like we did before that we would want to do as z stream.

**Comment by christianvogt** [comment]
Thanks @spadgett any content changes should be a followup PR.

File: quickstarts/explore-serverless-quickstart.yaml (hunk 10 above)
Comment (jhadvig) [reply quickstarts/explore-serverless-quickstart.yaml:14]: Makes sense, I was just a bit confused whats the right way how to use them.

**Comment by jhadvig** [comment]
@spadgett the PR looks good 👍 
One more thing what comes to my mind is if we want to add an e2e test in the console-operator, that would check if the QSs are created. But not sure if that makes sense, since the operator is not managing them. We can probably leave that to the future.

**Comment by spadgett** [comment]
> @spadgett the PR looks good 👍
> One more thing what comes to my mind is if we want to add an e2e test in the console-operator, that would check if the QSs are created. But not sure if that makes sense, since the operator is not managing them. We can probably leave that to the future.

Yeah, I was thinking about that. I wonder if this would be best as an e2e test in the console repo (since we run the console e2e tests here). Then we can confirm that they're showing up correctly in the UI.

**Pull request merged**